src/commands/tail/mod.rs

```diff
@@ -0,0 +1,220 @@
+use crate::http;
+use crate::settings::global_user::GlobalUser;
+use crate::settings::toml::Target;
+
+use cloudflare::endpoints::workers::{CreateTail, CreateTailParams, SendTailHeartbeat};
+use cloudflare::framework::HttpApiClientConfig;
+use cloudflare::framework::{async_api, async_api::ApiClient};
+
+use std::path::PathBuf;
+use std::process::Stdio;
+use std::str;
+use std::thread;
+use std::time::Duration;
+
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use log::log_enabled;
+use log::Level::Info;
+use regex::Regex;
+use reqwest;
+use tokio;
+use tokio::process::Command;
+use tokio::runtime::Runtime as TokioRuntime;
+
+pub fn start_tail(target: &Target, user: &GlobalUser) -> Result<(), failure::Error> {
+    let mut runtime = TokioRuntime::new()?;
+    // Note that we use eprintln!() throughout this file; this is because we want any
+    // helpful output to not be mixed with actual log JSON output, so we use this macro
+    // to print messages to stderr instead of stdout (where log output is printed).
+    eprintln!(
+        "Setting up log streaming from Worker \"{}\" to Wrangler. This may take a few seconds...",
+        target.name
+    );
+
+    // TODO: get rid of this gross clone. Maybe just default to static lifetime.
+    runtime.block_on(run_cloudflared_start_server(target.clone(), user.clone()))
+}
+
+// TODO: handle sigint to tear down tunnel, http server
+async fn run_cloudflared_start_server(
```

Comment:
nit: this name is a bit confusing, might just want to rename to `run` honestly

Reply:
i will be doing a bit of refactoring so renames are inevitable.